![https://raw.githubusercontent.com/marciolws/Curso_EBAC_Cientista_de_Dados/refs/heads/main/EBAC-media-utils/logo/ebac_logo-data_science.png](https://raw.githubusercontent.com/marciolws/Curso_EBAC_Cientista_de_Dados/refs/heads/main/EBAC-media-utils/logo/ebac_logo-data_science.png)

-----------------
<!-- # **Profissão: Cientista de Dados** -->
### **Módulo 25** | Cálculo | Exercício II

**Aluno:** [Marcio da Silva](https://www.linkedin.com/in/marciolws/)<br>
**Data:** 06 de outubro de 2024.

-----------------

## Cálculo - Tarefa 02

### 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  


In [1]:
import numpy as np

In [16]:
f = lambda x: x**3

def calc_area(a:int, b:int, func:callable, n_ret:int) -> float:
    delta = (b - a) / n_ret
    x_cols = np.linspace(a, b - delta, n_ret) + delta / 2
    y_cols = func(x_cols)

    print(f'Numero de retangulos = {n_ret}')
    print(f'Delta = {delta}')

    return sum(delta * y_cols)

In [23]:
calc_area(a=1, b=2, func=f, n_ret=1)

Numero de retangulos = 1
Delta = 1.0


np.float64(3.375)

In [26]:
calc_area(a=1, b=2, func=f, n_ret=20)

Numero de retangulos = 20
Delta = 0.05


np.float64(3.7490624999999995)

In [18]:
calc_area(a=1, b=2, func=f, n_ret=100)

Numero de retangulos = 100
Delta = 0.01


np.float64(3.7499624999999996)

### 2- Calcule essa área de forma analítica.

$$\int_{a}^{b} f(x) dx = F(b) - F(a)$$

$$ f(x) = x^3 \rightarrow F(x) = \frac{1}{4} * x^4 + C,$$

$$\int_{1}^{2} x^3 dx = F(2) - F(1) = \left(\frac{1}{4} * 2^4\right) - \left(\frac{1}{4} * 1^4\right) = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \textbf{3.75}$$